src/Node/AssertNode.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+/*
+ * This file is part of Twig.
+ *
+ * (c) Fabien Potencier
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Twig\Node;
+
+use Twig\Attribute\YieldReady;
+use Twig\Compiler;
+use Twig\Node\Expression\AbstractExpression;
+
+/**
+ * Represents a assert node.
+ *
+ * @author Simon André <smn.andre@gmail.com>
+ */
+#[YieldReady]
+class AssertNode extends Node
+{
+    public function __construct(AbstractExpression $expr, int $lineno, ?string $tag = null)
+    {
+        parent::__construct(['expr' => $expr], [], $lineno, $tag);
+    }
+
+    public function compile(Compiler $compiler): void
+    {
+        $compiler->addDebugInfo($this);
+        // TODO disable in production(?)
+
+        $compiler
+            ->write(' if (!')
+            ->subcompile($this->getNode('expr'))
+            ->raw(") {\n")
+            ->indent()
+            ->write("throw new RuntimeError('Failed');\n")
+            ->outdent()
+            ->raw("}\n")
+        ;
```

Comment:
What about using `assert()` function?  
With OPcache configuration, you can literally remove `assert()` calls from the code... or here you planned to not write anything if production?

Reply:
That's an idea too !  I would say a node would be easier to parse / use by node visitors but i may be wrong here 😅 

> or here you planned to not write anything if production?

I planned... _nothing_ 🙃   the "idea" just emerged earlier tonight during the "var/type" discussion

Reply:
The suggestion is about the compiled code, not about the Twig API

Actually, to be able to use PHP's `assert` function which is a no-op at runtime in production mode, it needs to be a Twig tag and not Twig function usable in an expression.

Reply:
👍